data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/VariableExpander.java

```diff
@@ -33,26 +42,46 @@ public VariableExpander(
             final ObjectMapper objectMapper,
             final Set<PluginConfigValueTranslator> pluginConfigValueTranslators) {
         this.objectMapper = objectMapper;
-        patternPluginConfigValueTranslatorMap = pluginConfigValueTranslators.stream().collect(Collectors.toMap(
-                pluginConfigValueTranslator -> Pattern.compile(
-                        String.format(SECRETS_REFERENCE_PATTERN_STRING,
-                                pluginConfigValueTranslator.getPrefix(), VALUE_REFERENCE_KEY)),
-                Function.identity()));
+        patternPluginConfigValueTranslatorMap = new HashMap<>();
+        pluginConfigValueTranslators.forEach(pluginConfigValueTranslator -> {
+            if (!DEFAULT_DEPRECATED_PREFIX.equals(
+                    pluginConfigValueTranslator.getDeprecatedPrefix())) {
+                patternPluginConfigValueTranslatorMap.put(
+                        Pattern.compile(
+                                String.format(SECRETS_REFERENCE_PATTERN_STRING,
+                                        pluginConfigValueTranslator.getDeprecatedPrefix(), VALUE_REFERENCE_KEY)),
+                        pluginConfigValueTranslator);
+            }
+            patternPluginConfigValueTranslatorMap.put(
+                    Pattern.compile(
+                            String.format(SECRETS_REFERENCE_PATTERN_STRING,
+                                    pluginConfigValueTranslator.getPrefix(), VALUE_REFERENCE_KEY)),
+                    pluginConfigValueTranslator);
+        });
     }
 
     public <T> T translate(final JsonParser jsonParser, final Class<T> destinationType) throws IOException {
         if (JsonToken.VALUE_STRING.equals(jsonParser.currentToken())) {
             final String rawValue = jsonParser.getValueAsString();
-            return patternPluginConfigValueTranslatorMap.entrySet().stream()
-                    .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey().matcher(rawValue), entry.getValue()))
-                    .filter(entry -> entry.getKey().matches())
-                    .map(entry -> {
-                        final String valueReferenceKey = entry.getKey().group(VALUE_REFERENCE_KEY);
-                        return objectMapper.convertValue(
-                                entry.getValue().translate(valueReferenceKey), destinationType);
-                    })
-                    .findFirst()
-                    .orElseGet(() -> objectMapper.convertValue(rawValue, destinationType));
+            final Matcher referenceMatcher = REFERENCE_PATTERN.matcher(rawValue);
+            if (referenceMatcher.matches()) {
+                LOG.debug("Searching for PluginConfigValueTranslator to match prefix on the raw value: {}", rawValue);
```

Comment:
I think we should split the prefix first. Maybe look for the last `:` instead? It is hard to know exactly what we are looking for and what is missing without more clarity.

I think the ideal error/warn log would look like:

```
Unable to find a variable expansion for prefix "aws-secrets:". Available prefixes are: "aws:secrets:", "env:"
```